src/test/java/unit/uk/gov/ida/verifyserviceprovider/resources/GenerateAuthnRequestResourceTest.java

```diff
@@ -106,16 +114,16 @@ public void samlRequestIsBase64EncodedAuthnRequest() {
         }
     }
 
+
     @Test
     public void returns422ForBadJson() {
         Response response = resources.target("/generate-request")
-            .request()
-            .post(Entity.entity("{}", MediaType.APPLICATION_JSON_TYPE));
+                .request()
+                .post(Entity.entity(ImmutableMap.of("bad", "json"), MediaType.APPLICATION_JSON_TYPE));
         assertThat(response.getStatus()).isEqualTo(422);
-        assertThat(response.readEntity(ErrorMessage.class)).isEqualTo(new ErrorMessage(
-            422,
-            "levelOfAssurance may not be null")
-        );
+        ErrorMessage errorMessage = response.readEntity(ErrorMessage.class);
+        assertThat(errorMessage.getCode()).isEqualTo(422);
+        assertThat(errorMessage.getMessage()).startsWith("Unrecognized field \"bad\"");
     }
```

Comment:
Shouldn't we be returning 422 in the "{}" case too? Or that be valid and not throw a 500.

Reply:
{} should be valid, but no body should also be valid - I don't think that's being tested.